Add documentation related to new settings for segment upload timeout (#7330)
opensearch-project/documentation-website: _install-and-configure/configuring-opensearch/index-settings.md

- `cluster.remote_store.index.segment_metadata.retention.max_count` (Integer): Controls the minimum number of metadata files to keep in the segment repository on a remote store. A value below `1` disables the deletion of stale segment metadata files. Default is `10`.

> - `cluster.remote_store.segment.transfer_timeout` (Time unit): Controls the maximum amount of time to wait for all new segments post refresh to be uploaded to remote store. If the upload doesnt complete in specified time it throws a SegmentUploadFailedException. Default is `30m`. It has a minimum constraint of `10m`.

Reviewer: ```suggestion
- `cluster.remote_store.segment.transfer_timeout` (Time unit): Controls the maximum amount of time to wait for all new segments to update post refresh to the remote store. If the upload does not complete in a specified time it throws a `SegmentUploadFailedException` error. Default is `30m`. It has a minimum constraint of `10m`.
```

> - `cluster.remote_store.segment.transfer_timeout` (Time unit): Controls the maximum amount of time to wait for all new segments to update post refresh to the remote store. If the upload does not complete in a specified time it throws a `SegmentUploadFailedException` error. Default is `30m`. It has a minimum constraint of `10m`.

Reviewer 2: ```suggestion
- `cluster.remote_store.segment.transfer_timeout` (Time unit): Controls the maximum amount of time to wait for all new segments to update after refresh to the remote store. If the upload does not complete within a specified amount of time, it throws a `SegmentUploadFailedException` error. Default is `30m`. It has a minimum constraint of `10m`.
```
